File: ALittle/Module/ALittleIDE/Other/GameLibrary/Std/JSScript/Std/Loop/LoopFrame.js
{
if (typeof ALittle === "undefined") window.ALittle = {};


if (ALittle.LoopObject === undefined) throw new Error(" extends class:ALittle.LoopObject is undefined");
ALittle.LoopFrame = JavaScript.Class(ALittle.LoopObject, {
	Ctor : function(func) {
		this._func = func;
		this._force_completed = false;
		if (this._func === undefined) {
			this._force_completed = true;
			ALittle.Log("LoopFrame create failed:function is nil or not a function");
			return;
		}
	},
	Reset : function() {
	},
	Completed : function() {
	},
	IsCompleted : function() {
		return this._force_completed;
	},
	SetCompleted : function() {
		this._force_completed = true;
	},
	SetTime : function(time) {
		return [time, true];
	},
	Update : function(frame_time) {
		this._func(frame_time);
	},
}, "ALittle.LoopFrame");

}

File: ALittle/Module/Std/Script/Std/Loop/LoopFrame.lua
-- ALittle Generate Lua
module("ALittle", package.seeall)

local ___rawset = rawset
local ___pairs = pairs
local ___ipairs = ipairs


assert(ALittle.LoopObject, " extends class:ALittle.LoopObject is nil")
LoopFrame = Lua.Class(ALittle.LoopObject, "ALittle.LoopFrame")

function LoopFrame:Ctor(func)
	___rawset(self, "_func", func)
	___rawset(self, "_force_completed", false)
	if self._func == nil then
		___rawset(self, "_force_completed", true)
		Log("LoopFrame create failed:function is nil or not a function")
		return
	end
end

function LoopFrame:Reset()
end

function LoopFrame:Completed()
end

function LoopFrame:IsCompleted()
	return self._force_completed
end

function LoopFrame:SetCompleted()
	self._force_completed = true
end

function LoopFrame:SetTime(time)
	return time, true
end

function LoopFrame:Update(frame_time)
	self._func(frame_time)
end

